WordPress/Classes/Utility/BuildInformation/FeatureFlag.swift

```diff
@@ -39,3 +47,98 @@ class Feature: NSObject {
         return feature.enabled
     }
 }
+
+
+extension FeatureFlag: CustomStringConvertible {
+    /// Descriptions used to display the feature flag override menu in debug builds
+    var description: String {
+        switch self {
+        case .exampleFeature:
+            return "Example feature"
+        case .jetpackDisconnect:
+            return "Jetpack disconnect"
+        case .domainCredit:
+            return "Use domain credits"
+        case .signInWithApple:
+            return "Sign in with Apple"
+        case .postScheduling:
+            return "Post scheduling improvements"
+        case .debugMenu:
+            return "Debug menu"
+        }
+    }
+}
+
+// MARK: - Overriding Feature Flags
+
+// Protocol allows easier unit testing, so we can implement mock
+// feature flags to use in test cases.
+//
+protocol OverrideableFlag: CustomStringConvertible {
+    var enabled: Bool { get }
+    var canOverride: Bool { get }
+}
+
+extension FeatureFlag: OverrideableFlag {
+    var canOverride: Bool {
+        switch self {
+        case .debugMenu, .exampleFeature:
+            return false
+        default:
+            return true
+        }
+    }
+}
+
+/// Used to override values for feature flags at runtime in debug builds
+///
+struct FeatureFlagOverrideStore {
```

Comment:
What do you think about extracting the `FeatureFlagOverrideStore` to its own file?

Reply:
Hmm, possibly? I left it in there as I felt it wasn't too big and obviously they're quite closely related. I think if we did extract it, I'd like to leave the `description` implementation in `FeatureFlag.swift`, as I think it's best to keep that near the initial definition of the FeatureFlag enum. Does that make sense? Which one would you prefer I change?

Reply:
We should definitely keep the `description` in `FeatureFlag.swift`!

I just thought about moving the `FeatureFlagOverrideStore` to another file to keep `FeatureFlag.swift` cleaner — just the flags and the descriptions.

But this is not a big deal, tbh. :)